pandas-stubs/core/frame.pyi

```diff
@@ -48,7 +48,10 @@ class DataFrame:
     def __getitem__(self, idx: _str) -> Series: ...
     @overload
     def __getitem__(
-        self, idx: Union[Series, Tuple[_str, ...], List[_str], Index[_str], _np.ndarray[_np.str_]]
+        self,
+        idx: Union[
+            Series, Tuple[_str, ...], List[_str], Index[_str], _np.ndarray[_np.str_], DataFrame
```

Comment:
Do you have an example for when you need to pass a `DataFrame` here? Does the `DataFrame` contain strings?

Reply:
Unfortunately, I can't put the code I'm working on up on github, but I needed to add that as an option. The data frames I'm working with are... nested in weird ways that I don't quite understand yet. I'll have to bug the rest of the team about why they're doing things this way.

Reply:
`frame = frame[frame['Framename'].to_frame().isin(setOfFrames)]`

Like I said, weird.

Reply:
I see... I'm guessing `frame['Framename'].to_frame().isin(setOfFrames)` returns some kind of mask? so, a dataframe where each element is either True or False?

If it is a mask, then you should use `.loc` instead [as the documentation recommends](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.loc.html). This is a typical example from the documentation:

```python
df.loc[df['shield'] > 6]
```

I want to claim that we support this already, but looking at the code, we actually don't support it. Let me add a PR for that.

I will think about this a bit more and reply to you tomorrow. The other changes in this PR can definitely be merged.

Reply:
I'll be able to ask my coworkers in person about this on Monday, so I'll have a more solid grasp of why things are the way they are after that.

Reply:
After rereading the code - and I still need to confirm this with the team - I believe this line culls out all the values which aren't also in the setOfFrames.

Reply:
I reverted the addition and used .loc in its place.